utils/cagg_refresh_producer_consumer/consumer.sql

```diff
@@ -0,0 +1,176 @@
+
+DROP PROCEDURE IF EXISTS _timescaledb_additional.task_refresh_continuous_aggregate_incremental_runner;
+CREATE PROCEDURE _timescaledb_additional.task_refresh_continuous_aggregate_incremental_runner (
+    job_id int,
+    config jsonb
+) LANGUAGE plpgsql AS $BODY$
+DECLARE
+    max_runtime interval := (config->>'max_runtime')::interval;
+    global_start_time timestamptz := pg_catalog.clock_timestamp();
+    global_end_time timestamptz;
+    app_name text;
+BEGIN
+    max_runtime := coalesce(max_runtime, interval '6 hours');
+    global_end_time := global_start_time + max_runtime;
+
+    WHILE pg_catalog.clock_timestamp() < global_end_time LOOP
+        SET search_path TO 'pg_catalog,pg_temp';
+        SET lock_timeout TO '3s';
+        SET application_name TO 'cagg incremental refresh consumer - idle';
+
+        -- Prevent a hot loop
+        PERFORM pg_catalog.pg_sleep(1.0);
+
+        -- By serializing the picking items from the queue, we prevent some race conditions.
+        LOCK TABLE _timescaledb_additional.incremental_continuous_aggregate_refreshes
+            IN ACCESS EXCLUSIVE MODE;
```

Comment:
My be worthy of a comment, but the tx that takes this lock is always short lived.
It doesn't do heavy lifting, it populates `worker_pid` and commits.

Reply:
Do we really need this?? We're using FOR UPDATE when picking an item from the queue